src/org/daisy/dotify/formatter/impl/page/PageSequenceBuilder2.java

```diff
@@ -322,6 +322,10 @@ private PageImpl nextPageInner(
             current.getPageTemplate().validateAndAnalyzeFooter()
         );
 
+        // At the beginning here before we start printing the page for this iteration we will set
+        // the value topOfPage and because this is a mutable context value we need to build a
+        // new object. This value will be changed later on when we aren't at the top of
+        // the page anymore. This is signified by that we have written a full row group.
```

Comment:
To be precise, it should be mentioned that setting `.topOfPage(true)` here will in the general case _NOT_ be correct (according to the definition of the OBFL variable `$starts-at-top-of-page`), however in the specific case when the value is read (when evaluating a `display-when="(! $starts-at-top-of-page)"`, refer to `addRows()` below) it _WILL_ be correct under all the assumptions made.